docs/proposals/hpa/centralized-hpa.md

```diff
@@ -0,0 +1,183 @@
+1. If the system does not meet the first requirement, some clusters' services may become overloaded while others remain underloaded.      
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. If the HPA is propagated to member clusters, there will be conflicts with Karmada control plane. So users couldn't propagate HPA from Karmada control plane.
+2. The HPA used in a single cluster can be seamlessly migrated to multiple clusters' centralized HPA.
+
+## Design Details
+### Architecture
+For different scenarios, the architecture may be different:  
+HPA with CPU/Memory metrics:  
+![K8s native metrics](./statics/centralizedhpa-native-metrics.png)  
+
+HPA with custom metrics:  
+![Custom metrics](./statics/centralizedhpa-custom-metrics.png)  
+
+HPA with managed metrics store:  
+![Managed metrics store](./statics/managed-metrics-store.png)  
+
+From the above architecture, we can get following information:
+* Centralized HPA controller is the original HPA controller in `karmada-kube-controller-manager` component, which will query metrics from `karmada-apiserver` with metrics API and scale the deployments.
+* `karmada-metrics-adaptor` is responsible for query metrics from member clusters and do aggregation.
+* Centralized HPA only responsible for scaling instances but doesn't care where the instances will be scheduled.
+* The HPA version should be compatible with `karmada-kube-controller-manager` version.
+
+For better user experience, centralized HPA API is same with original HPA API.
+
+### Components change
+
+#### karmada-kube-controller-manager  
+In order to make HPA scaling work, HPA controller should be enablded in `karmada-kube-controller-manager` as follows:
+```yaml
+- command:
+  - kube-controller-manager
+  - ...
+  - --controllers=xxx,horizontalpodautoscaling
+  - ...
+```
+
+#### karmada-metrics-adaptor
+To query metrics from member clusters, Karmada introduces a new component called `karmada-metrics-adaptor` which is responsible for the following tasks:  
+* Registering the metrics API service including metrics.k8s.io, custom.metrics.k8s.io, and external.metrics.k8s.io.
+* Listing and watching ResourceBindings to know where workloads are scheduled. When a query is received from apiserver, the `karmada-metrics-adaptor` will query the member clusters and perform aggregation.
+
+For different PropagationPolicy, the aggregation will be different:
+* For divided method, `karmada-metrics-adaptor` will return the metrics found directly from the member clusters.
```

Comment:
For divide method, we regard the distributed replicas as one application. So we might do some calculation after we get all metrics from member clusters. Here can be explained in detail.